packages/vuetify/src/components/VDataTable/VDataTable.ts

```diff
@@ -198,6 +202,19 @@ export default VDataIterator.extend({
     calcWidths () {
       this.widths = Array.from(this.$el.querySelectorAll('th')).map(e => e.clientWidth)
     },
+    createItemClass (classes: string|string[]|Record<string, boolean>) {
```

Comment:
null/undefined should be allowed too 
Extract type and add to `types/index.d.ts`

It could be:

```ts
export type XXX = null|undefined|string|string[]|Record<string, boolean>
export type XXX_PROP = (item: any) => XXX
```
or

```ts
export type XXX_PROP = (item: any) => null|undefined|string|string[]|Record<string, boolean>

// usage
createItemClass (classes: ReturnType<XXX_PROP>) {...}
```